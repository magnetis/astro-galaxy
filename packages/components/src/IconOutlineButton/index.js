import PropTypes from 'prop-types';
import styled from 'styled-components';
import OutlineButton from '../OutlineButton';

const iconColor = props => ({
  'outline.uranus': `
    color: ${props.theme.colors.uranus500};
  `,
  'outline.earth': `
    color: ${props.theme.colors.earth600};
  `,
  'outline.venus': `
    color: ${props.theme.colors.venus400};
  `,
  'outline.mars': `
    color: ${props.theme.colors.mars500};
  `,
});

const hoverStyles = props => ({
  'outline.uranus': `
    color: ${props.theme.colors.space100};
  `,
  'outline.earth': `
    color: ${props.theme.colors.moon900};
  `,
  'outline.venus': `
    color: ${props.theme.colors.space100};
  `,
  'outline.mars': `
    color: ${props.theme.colors.space100};
  `,
});

const iconSize = () => ({
  small: `
    width: 24px;
    height: 24px;
  `,
  medium: `
    width: 32px;
    height: 32px;
  `,
  large: `
    width: 40px;
    height: 40px;
  `,
});

const buttonPadding = () => ({
  small: `
    padding: 5px 2px;
  `,
  medium: `
    padding: 8px 6px;
  `,
  large: `
    padding: 11px 9px;
  `,
});

const iconlabelMargin = props => {
  if (props.iconlabel && !props.iconlabelRight) {
    return '-4px 9px -3px 0';
  }

  if (props.iconlabel && props.iconlabelRight) {
    return '-4px 0 -3px 9px';
  }

  return '-4px 0 -3px';
};

const IconOutlineButton = styled(OutlineButton)`
  & svg {
    transition: all 0.3s ease;
    vertical-align: middle;
    margin: ${iconlabelMargin};
    ${props => iconColor(props)[props.variant]};
    ${props => iconSize()[props.buttonSize]};
  }

  ${props => !props.iconlabel && buttonPadding()[props.buttonSize]};

  & :disabled {
    svg {
      color: ${props => props.theme.colors.moon200};
    }
  }

  :hover:not(:disabled) svg,
  :focus:not(:disabled) svg,
  :active:not(:disabled) svg {
    ${props => hoverStyles(props)[props.variant]}
  }
`;

IconOutlineButton.displayName = 'IconOutlineButton';

IconOutlineButton.propTypes = {
  variant: PropTypes.oneOf(['outline.uranus', 'outline.earth', 'outline.venus', 'outline.mars']),
  buttonSize: PropTypes.oneOf(['small', 'medium', 'large']),
  iconlabel: PropTypes.bool,
  iconlabelRight: PropTypes.bool,
};

IconOutlineButton.defaultProps = {
  variant: 'outline.uranus',
  buttonSize: 'medium',
  iconlabel: false,
  iconlabelRight: false,
};

export default IconOutlineButton;
