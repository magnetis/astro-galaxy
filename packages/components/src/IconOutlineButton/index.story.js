import React from 'react';
import { storiesOf } from '@storybook/react';
import IconOutlineButton from './index.js';
import { IconCalendar } from '@magnetis/astro-galaxy-icons';

storiesOf('icon outline buttons', module)
  .add('icon button default', () => (
    <IconOutlineButton>
      <IconCalendar />
    </IconOutlineButton>
  ))
  .add('icon outline buttons variants', () => (
    <>
      <IconOutlineButton variant="outline.uranus">
        <IconCalendar />
      </IconOutlineButton>
      <IconOutlineButton variant="outline.earth">
        <IconCalendar />
      </IconOutlineButton>
      <IconOutlineButton variant="outline.venus">
        <IconCalendar />
      </IconOutlineButton>
      <IconOutlineButton variant="outline.mars">
        <IconCalendar />
      </IconOutlineButton>
      <IconOutlineButton disabled>
        <IconCalendar />
      </IconOutlineButton>
    </>
  ));

storiesOf('icon outline button sizes', module).add('icon outline buttons sizes', () => (
  <>
    <IconOutlineButton buttonSize="small">
      <IconCalendar />
    </IconOutlineButton>
    <IconOutlineButton buttonSize="medium">
      <IconCalendar />
    </IconOutlineButton>
    <IconOutlineButton buttonSize="large">
      <IconCalendar />
    </IconOutlineButton>
  </>
));

storiesOf('iconlabel outline button', module)
  .add('iconlabel', () => (
    <>
      <IconOutlineButton variant="outline.uranus" iconlabel>
        <IconCalendar />
        uranus
      </IconOutlineButton>
      <IconOutlineButton variant="outline.earth" iconlabel>
        <IconCalendar />
        earth
      </IconOutlineButton>
      <IconOutlineButton variant="outline.venus" iconlabel>
        <IconCalendar />
        venus
      </IconOutlineButton>
      <IconOutlineButton variant="outline.mars" iconlabel>
        <IconCalendar />
        mars
      </IconOutlineButton>
      <IconOutlineButton iconlabel disabled>
        <IconCalendar />
        disabled
      </IconOutlineButton>
    </>
  ))
  .add('iconlabelRight', () => (
    <>
      <IconOutlineButton variant="outline.uranus" iconlabel iconlabelRight>
        uranus
        <IconCalendar />
      </IconOutlineButton>
      <IconOutlineButton variant="outline.earth" iconlabel iconlabelRight>
        earth
        <IconCalendar />
      </IconOutlineButton>
      <IconOutlineButton variant="outline.venus" iconlabel iconlabelRight>
        venus
        <IconCalendar />
      </IconOutlineButton>
      <IconOutlineButton variant="outline.mars" iconlabel iconlabelRight>
        mars
        <IconCalendar />
      </IconOutlineButton>
      <IconOutlineButton iconlabel disabled>
        disabled
        <IconCalendar />
      </IconOutlineButton>
    </>
  ))
  .add('iconlabel button sizes', () => (
    <>
      <IconOutlineButton buttonSize="small" iconlabel>
        <IconCalendar />
        uranus
      </IconOutlineButton>
      <IconOutlineButton buttonSize="medium" iconlabel>
        <IconCalendar />
        uranus
      </IconOutlineButton>
      <IconOutlineButton buttonSize="large" iconlabel>
        <IconCalendar />
        uranus
      </IconOutlineButton>
    </>
  ));
