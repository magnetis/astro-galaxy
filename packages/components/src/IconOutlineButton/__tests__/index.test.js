import React from 'react';
import { theme } from '@magnetis/astro-galaxy-core';
import { IconCalendar } from '@magnetis/astro-galaxy-icons';
import IconOutlineButton from '../index';

describe('IconOutlineButton', () => {
  it('should render an icon button with calendar icon', () => {
    const iconButton = rendererCreateWithTheme(
      <IconOutlineButton>
        <IconCalendar />
      </IconOutlineButton>
    ).toJSON();

    expect(iconButton).toMatchSnapshot();
  });

  it('should change padding when icon button size is small', () => {
    const iconButton = rendererCreateWithTheme(
      <IconOutlineButton buttonSize="small">
        <IconCalendar />
      </IconOutlineButton>
    ).toJSON();

    expect(iconButton).toMatchSnapshot();
    expect(iconButton).toHaveStyleRule('padding', '5px 2px');
  });

  it('should change padding when icon button size is large', () => {
    const iconButton = rendererCreateWithTheme(
      <IconOutlineButton buttonSize="large">
        <IconCalendar />
      </IconOutlineButton>
    ).toJSON();

    expect(iconButton).toMatchSnapshot();
    expect(iconButton).toHaveStyleRule('padding', '11px 9px');
  });

  it('should border-color when variant is outline.earth', () => {
    const iconButton = rendererCreateWithTheme(
      <IconOutlineButton variant="outline.earth">
        <IconCalendar />
      </IconOutlineButton>
    ).toJSON();

    expect(iconButton).toMatchSnapshot();
    expect(iconButton).toHaveStyleRule('border-color', theme.colors.earth600);
  });

  it('should change border-color when variant is outline.venus', () => {
    const iconButton = rendererCreateWithTheme(
      <IconOutlineButton variant="outline.venus">
        <IconCalendar />
      </IconOutlineButton>
    ).toJSON();

    expect(iconButton).toMatchSnapshot();
    expect(iconButton).toHaveStyleRule('border-color', theme.colors.venus400);
  });

  it('should change border-color when variant is outline.mars', () => {
    const iconButton = rendererCreateWithTheme(
      <IconOutlineButton variant="outline.mars">
        <IconCalendar />
      </IconOutlineButton>
    ).toJSON();

    expect(iconButton).toMatchSnapshot();
    expect(iconButton).toHaveStyleRule('border-color', theme.colors.mars500);
  });
});
