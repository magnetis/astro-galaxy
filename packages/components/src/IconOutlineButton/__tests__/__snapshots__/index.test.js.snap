// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconOutlineButton should border-color when variant is outline.earth 1`] = `
.c1 {
  fill: currentcolor;
  color: #32383c;
}

.c0 {
  cursor: pointer;
  white-space: nowrap;
  text-align: center;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  overflow: hidden;
  border-width: 2px;
  border-style: solid;
  -webkit-text-decoration: none;
  text-decoration: none;
  font: normal normal 600 normal 16px/1.5 Poppins,sans-serif;
  display: inline-block;
  color: #469d39;
  border-color: #469d39;
  padding: 8px 38px;
  font-size: 16px;
  line-height: 28px;
  border-radius: 24px;
  background-color: transparent;
  padding: 8px 6px;
}

.c0:disabled {
  cursor: not-allowed;
  background-color: #e9eef2;
  border-color: #e9eef2;
  color: #ffffff;
}

.c0:focus {
  outline: none;
}

.c0:disabled {
  cursor: not-allowed;
  background-color: transparent;
  border-color: #e9eef2;
  color: #e9eef2;
}

.c0:hover:not(:disabled),
.c0:focus:not(:disabled) {
  background-color: #88e967;
  border-color: #88e967;
  color: #32383c;
}

.c0:active:not(:disabled) {
  background-color: #56b53f;
  border-color: #56b53f;
}

.c0:focus {
  outline: none;
}

.c0 svg {
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  vertical-align: middle;
  margin: -4px 0 -3px;
  color: #469d39;
  width: 32px;
  height: 32px;
}

.c0:disabled svg {
  color: #e9eef2;
}

.c0:hover:not(:disabled) svg,
.c0:focus:not(:disabled) svg,
.c0:active:not(:disabled) svg {
  color: #32383c;
}

@media (prefers-reduced-motion:reduce) {

}

<button
  className="Button-tgj28e-0 OutlineButton-sc-9jysz0-0 c0"
  display="inline-block"
>
  <svg
    className="c1"
    color="moon900"
    height={24}
    viewBox="0 0 32 32"
    width={24}
  >
    <defs>
      <path
        d="M16 4H8v2a1 1 0 1 1-2 0V4H3a1 1 0 0 0-1 1v4.111h20V5a1 1 0 0 0-1-1h-3v2a1 1 0 0 1-2 0V4zm1 0v2-2zm1-2v2h1V2h2a3 3 0 0 1 3 3v18a3 3 0 0 1-3 3H3a3 3 0 0 1-3-3V5a3 3 0 0 1 3-3h2v2h1V1a1 1 0 1 1 2 0v3h1V2h6v2h1V1a1 1 0 0 1 2 0v1zm-1 0V1v1zM7 4v2-2zm0-2V1v1zm15 9.111H2V23a1 1 0 0 0 1 1h18a1 1 0 0 0 1-1V11.111zM7 17a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm5-4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm5-4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2z"
        id="icon-calendar-a"
      />
    </defs>
    <g
      id="icon/secondary/calendar"
    >
      <g
        id="color/earth-400"
        transform="translate(4 3)"
      >
        <mask
          id="icon-calendar-b"
        >
          <use
            xlinkHref="#icon-calendar-a"
          />
        </mask>
        <use
          id="Mask"
          xlinkHref="#icon-calendar-a"
        />
        <g
          id="color/moon-900"
          mask="url(#icon-calendar-b)"
        >
          <path
            d="M0 0h32v32H0z"
            id="Rectangle"
            transform="translate(-4 -3)"
          />
        </g>
      </g>
    </g>
  </svg>
</button>
`;

exports[`IconOutlineButton should change border-color when variant is outline.mars 1`] = `
.c1 {
  fill: currentcolor;
  color: #32383c;
}

.c0 {
  cursor: pointer;
  white-space: nowrap;
  text-align: center;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  overflow: hidden;
  border-width: 2px;
  border-style: solid;
  -webkit-text-decoration: none;
  text-decoration: none;
  font: normal normal 600 normal 16px/1.5 Poppins,sans-serif;
  display: inline-block;
  color: #d45459;
  border-color: #d45459;
  padding: 8px 38px;
  font-size: 16px;
  line-height: 28px;
  border-radius: 24px;
  background-color: transparent;
  padding: 8px 6px;
}

.c0:disabled {
  cursor: not-allowed;
  background-color: #e9eef2;
  border-color: #e9eef2;
  color: #ffffff;
}

.c0:focus {
  outline: none;
}

.c0:disabled {
  cursor: not-allowed;
  background-color: transparent;
  border-color: #e9eef2;
  color: #e9eef2;
}

.c0:hover:not(:disabled),
.c0:focus:not(:disabled) {
  background-color: #dd676b;
  border-color: #dd676b;
  color: #ffffff;
}

.c0:active:not(:disabled) {
  background-color: #ac3c40;
  border-color: #ac3c40;
}

.c0:focus {
  outline: none;
}

.c0 svg {
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  vertical-align: middle;
  margin: -4px 0 -3px;
  color: #d45459;
  width: 32px;
  height: 32px;
}

.c0:disabled svg {
  color: #e9eef2;
}

.c0:hover:not(:disabled) svg,
.c0:focus:not(:disabled) svg,
.c0:active:not(:disabled) svg {
  color: #ffffff;
}

@media (prefers-reduced-motion:reduce) {

}

<button
  className="Button-tgj28e-0 OutlineButton-sc-9jysz0-0 c0"
  display="inline-block"
>
  <svg
    className="c1"
    color="moon900"
    height={24}
    viewBox="0 0 32 32"
    width={24}
  >
    <defs>
      <path
        d="M16 4H8v2a1 1 0 1 1-2 0V4H3a1 1 0 0 0-1 1v4.111h20V5a1 1 0 0 0-1-1h-3v2a1 1 0 0 1-2 0V4zm1 0v2-2zm1-2v2h1V2h2a3 3 0 0 1 3 3v18a3 3 0 0 1-3 3H3a3 3 0 0 1-3-3V5a3 3 0 0 1 3-3h2v2h1V1a1 1 0 1 1 2 0v3h1V2h6v2h1V1a1 1 0 0 1 2 0v1zm-1 0V1v1zM7 4v2-2zm0-2V1v1zm15 9.111H2V23a1 1 0 0 0 1 1h18a1 1 0 0 0 1-1V11.111zM7 17a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm5-4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm5-4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2z"
        id="icon-calendar-a"
      />
    </defs>
    <g
      id="icon/secondary/calendar"
    >
      <g
        id="color/earth-400"
        transform="translate(4 3)"
      >
        <mask
          id="icon-calendar-b"
        >
          <use
            xlinkHref="#icon-calendar-a"
          />
        </mask>
        <use
          id="Mask"
          xlinkHref="#icon-calendar-a"
        />
        <g
          id="color/moon-900"
          mask="url(#icon-calendar-b)"
        >
          <path
            d="M0 0h32v32H0z"
            id="Rectangle"
            transform="translate(-4 -3)"
          />
        </g>
      </g>
    </g>
  </svg>
</button>
`;

exports[`IconOutlineButton should change border-color when variant is outline.venus 1`] = `
.c1 {
  fill: currentcolor;
  color: #32383c;
}

.c0 {
  cursor: pointer;
  white-space: nowrap;
  text-align: center;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  overflow: hidden;
  border-width: 2px;
  border-style: solid;
  -webkit-text-decoration: none;
  text-decoration: none;
  font: normal normal 600 normal 16px/1.5 Poppins,sans-serif;
  display: inline-block;
  color: #e078d6;
  border-color: #e078d6;
  padding: 8px 38px;
  font-size: 16px;
  line-height: 28px;
  border-radius: 24px;
  background-color: transparent;
  padding: 8px 6px;
}

.c0:disabled {
  cursor: not-allowed;
  background-color: #e9eef2;
  border-color: #e9eef2;
  color: #ffffff;
}

.c0:focus {
  outline: none;
}

.c0:disabled {
  cursor: not-allowed;
  background-color: transparent;
  border-color: #e9eef2;
  color: #e9eef2;
}

.c0:hover:not(:disabled),
.c0:focus:not(:disabled) {
  background-color: #e78ade;
  border-color: #e78ade;
  color: #ffffff;
}

.c0:active:not(:disabled) {
  background-color: #c763ba;
  border-color: #c763ba;
}

.c0:focus {
  outline: none;
}

.c0 svg {
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  vertical-align: middle;
  margin: -4px 0 -3px;
  color: #e078d6;
  width: 32px;
  height: 32px;
}

.c0:disabled svg {
  color: #e9eef2;
}

.c0:hover:not(:disabled) svg,
.c0:focus:not(:disabled) svg,
.c0:active:not(:disabled) svg {
  color: #ffffff;
}

@media (prefers-reduced-motion:reduce) {

}

<button
  className="Button-tgj28e-0 OutlineButton-sc-9jysz0-0 c0"
  display="inline-block"
>
  <svg
    className="c1"
    color="moon900"
    height={24}
    viewBox="0 0 32 32"
    width={24}
  >
    <defs>
      <path
        d="M16 4H8v2a1 1 0 1 1-2 0V4H3a1 1 0 0 0-1 1v4.111h20V5a1 1 0 0 0-1-1h-3v2a1 1 0 0 1-2 0V4zm1 0v2-2zm1-2v2h1V2h2a3 3 0 0 1 3 3v18a3 3 0 0 1-3 3H3a3 3 0 0 1-3-3V5a3 3 0 0 1 3-3h2v2h1V1a1 1 0 1 1 2 0v3h1V2h6v2h1V1a1 1 0 0 1 2 0v1zm-1 0V1v1zM7 4v2-2zm0-2V1v1zm15 9.111H2V23a1 1 0 0 0 1 1h18a1 1 0 0 0 1-1V11.111zM7 17a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm5-4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm5-4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2z"
        id="icon-calendar-a"
      />
    </defs>
    <g
      id="icon/secondary/calendar"
    >
      <g
        id="color/earth-400"
        transform="translate(4 3)"
      >
        <mask
          id="icon-calendar-b"
        >
          <use
            xlinkHref="#icon-calendar-a"
          />
        </mask>
        <use
          id="Mask"
          xlinkHref="#icon-calendar-a"
        />
        <g
          id="color/moon-900"
          mask="url(#icon-calendar-b)"
        >
          <path
            d="M0 0h32v32H0z"
            id="Rectangle"
            transform="translate(-4 -3)"
          />
        </g>
      </g>
    </g>
  </svg>
</button>
`;

exports[`IconOutlineButton should change padding when icon button size is large 1`] = `
.c1 {
  fill: currentcolor;
  color: #32383c;
}

.c0 {
  cursor: pointer;
  white-space: nowrap;
  text-align: center;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  overflow: hidden;
  border-width: 2px;
  border-style: solid;
  -webkit-text-decoration: none;
  text-decoration: none;
  font: normal normal 600 normal 16px/1.5 Poppins,sans-serif;
  display: inline-block;
  color: #159ce4;
  border-color: #159ce4;
  padding: 12px 46px;
  font-size: 24px;
  line-height: 35px;
  border-radius: 32px;
  background-color: transparent;
  padding: 11px 9px;
}

.c0:disabled {
  cursor: not-allowed;
  background-color: #e9eef2;
  border-color: #e9eef2;
  color: #ffffff;
}

.c0:focus {
  outline: none;
}

.c0:disabled {
  cursor: not-allowed;
  background-color: transparent;
  border-color: #e9eef2;
  color: #e9eef2;
}

.c0:hover:not(:disabled),
.c0:focus:not(:disabled) {
  background-color: #3db5eb;
  border-color: #3db5eb;
  color: #ffffff;
}

.c0:active:not(:disabled) {
  background-color: #147ec1;
  border-color: #147ec1;
}

.c0:focus {
  outline: none;
}

.c0 svg {
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  vertical-align: middle;
  margin: -4px 0 -3px;
  color: #159ce4;
  width: 40px;
  height: 40px;
}

.c0:disabled svg {
  color: #e9eef2;
}

.c0:hover:not(:disabled) svg,
.c0:focus:not(:disabled) svg,
.c0:active:not(:disabled) svg {
  color: #ffffff;
}

@media (prefers-reduced-motion:reduce) {

}

<button
  className="Button-tgj28e-0 OutlineButton-sc-9jysz0-0 c0"
  display="inline-block"
>
  <svg
    className="c1"
    color="moon900"
    height={24}
    viewBox="0 0 32 32"
    width={24}
  >
    <defs>
      <path
        d="M16 4H8v2a1 1 0 1 1-2 0V4H3a1 1 0 0 0-1 1v4.111h20V5a1 1 0 0 0-1-1h-3v2a1 1 0 0 1-2 0V4zm1 0v2-2zm1-2v2h1V2h2a3 3 0 0 1 3 3v18a3 3 0 0 1-3 3H3a3 3 0 0 1-3-3V5a3 3 0 0 1 3-3h2v2h1V1a1 1 0 1 1 2 0v3h1V2h6v2h1V1a1 1 0 0 1 2 0v1zm-1 0V1v1zM7 4v2-2zm0-2V1v1zm15 9.111H2V23a1 1 0 0 0 1 1h18a1 1 0 0 0 1-1V11.111zM7 17a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm5-4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm5-4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2z"
        id="icon-calendar-a"
      />
    </defs>
    <g
      id="icon/secondary/calendar"
    >
      <g
        id="color/earth-400"
        transform="translate(4 3)"
      >
        <mask
          id="icon-calendar-b"
        >
          <use
            xlinkHref="#icon-calendar-a"
          />
        </mask>
        <use
          id="Mask"
          xlinkHref="#icon-calendar-a"
        />
        <g
          id="color/moon-900"
          mask="url(#icon-calendar-b)"
        >
          <path
            d="M0 0h32v32H0z"
            id="Rectangle"
            transform="translate(-4 -3)"
          />
        </g>
      </g>
    </g>
  </svg>
</button>
`;

exports[`IconOutlineButton should change padding when icon button size is small 1`] = `
.c1 {
  fill: currentcolor;
  color: #32383c;
}

.c0 {
  cursor: pointer;
  white-space: nowrap;
  text-align: center;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  overflow: hidden;
  border-width: 2px;
  border-style: solid;
  -webkit-text-decoration: none;
  text-decoration: none;
  font: normal normal 600 normal 16px/1.5 Poppins,sans-serif;
  display: inline-block;
  color: #159ce4;
  border-color: #159ce4;
  padding: 5px 28px;
  font-size: 12px;
  line-height: 18px;
  border-radius: 16px;
  background-color: transparent;
  padding: 5px 2px;
}

.c0:disabled {
  cursor: not-allowed;
  background-color: #e9eef2;
  border-color: #e9eef2;
  color: #ffffff;
}

.c0:focus {
  outline: none;
}

.c0:disabled {
  cursor: not-allowed;
  background-color: transparent;
  border-color: #e9eef2;
  color: #e9eef2;
}

.c0:hover:not(:disabled),
.c0:focus:not(:disabled) {
  background-color: #3db5eb;
  border-color: #3db5eb;
  color: #ffffff;
}

.c0:active:not(:disabled) {
  background-color: #147ec1;
  border-color: #147ec1;
}

.c0:focus {
  outline: none;
}

.c0 svg {
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  vertical-align: middle;
  margin: -4px 0 -3px;
  color: #159ce4;
  width: 24px;
  height: 24px;
}

.c0:disabled svg {
  color: #e9eef2;
}

.c0:hover:not(:disabled) svg,
.c0:focus:not(:disabled) svg,
.c0:active:not(:disabled) svg {
  color: #ffffff;
}

@media (prefers-reduced-motion:reduce) {

}

<button
  className="Button-tgj28e-0 OutlineButton-sc-9jysz0-0 c0"
  display="inline-block"
>
  <svg
    className="c1"
    color="moon900"
    height={24}
    viewBox="0 0 32 32"
    width={24}
  >
    <defs>
      <path
        d="M16 4H8v2a1 1 0 1 1-2 0V4H3a1 1 0 0 0-1 1v4.111h20V5a1 1 0 0 0-1-1h-3v2a1 1 0 0 1-2 0V4zm1 0v2-2zm1-2v2h1V2h2a3 3 0 0 1 3 3v18a3 3 0 0 1-3 3H3a3 3 0 0 1-3-3V5a3 3 0 0 1 3-3h2v2h1V1a1 1 0 1 1 2 0v3h1V2h6v2h1V1a1 1 0 0 1 2 0v1zm-1 0V1v1zM7 4v2-2zm0-2V1v1zm15 9.111H2V23a1 1 0 0 0 1 1h18a1 1 0 0 0 1-1V11.111zM7 17a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm5-4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm5-4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2z"
        id="icon-calendar-a"
      />
    </defs>
    <g
      id="icon/secondary/calendar"
    >
      <g
        id="color/earth-400"
        transform="translate(4 3)"
      >
        <mask
          id="icon-calendar-b"
        >
          <use
            xlinkHref="#icon-calendar-a"
          />
        </mask>
        <use
          id="Mask"
          xlinkHref="#icon-calendar-a"
        />
        <g
          id="color/moon-900"
          mask="url(#icon-calendar-b)"
        >
          <path
            d="M0 0h32v32H0z"
            id="Rectangle"
            transform="translate(-4 -3)"
          />
        </g>
      </g>
    </g>
  </svg>
</button>
`;

exports[`IconOutlineButton should render an icon button with calendar icon 1`] = `
.c1 {
  fill: currentcolor;
  color: #32383c;
}

.c0 {
  cursor: pointer;
  white-space: nowrap;
  text-align: center;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  overflow: hidden;
  border-width: 2px;
  border-style: solid;
  -webkit-text-decoration: none;
  text-decoration: none;
  font: normal normal 600 normal 16px/1.5 Poppins,sans-serif;
  display: inline-block;
  color: #159ce4;
  border-color: #159ce4;
  padding: 8px 38px;
  font-size: 16px;
  line-height: 28px;
  border-radius: 24px;
  background-color: transparent;
  padding: 8px 6px;
}

.c0:disabled {
  cursor: not-allowed;
  background-color: #e9eef2;
  border-color: #e9eef2;
  color: #ffffff;
}

.c0:focus {
  outline: none;
}

.c0:disabled {
  cursor: not-allowed;
  background-color: transparent;
  border-color: #e9eef2;
  color: #e9eef2;
}

.c0:hover:not(:disabled),
.c0:focus:not(:disabled) {
  background-color: #3db5eb;
  border-color: #3db5eb;
  color: #ffffff;
}

.c0:active:not(:disabled) {
  background-color: #147ec1;
  border-color: #147ec1;
}

.c0:focus {
  outline: none;
}

.c0 svg {
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  vertical-align: middle;
  margin: -4px 0 -3px;
  color: #159ce4;
  width: 32px;
  height: 32px;
}

.c0:disabled svg {
  color: #e9eef2;
}

.c0:hover:not(:disabled) svg,
.c0:focus:not(:disabled) svg,
.c0:active:not(:disabled) svg {
  color: #ffffff;
}

@media (prefers-reduced-motion:reduce) {

}

<button
  className="Button-tgj28e-0 OutlineButton-sc-9jysz0-0 c0"
  display="inline-block"
>
  <svg
    className="c1"
    color="moon900"
    height={24}
    viewBox="0 0 32 32"
    width={24}
  >
    <defs>
      <path
        d="M16 4H8v2a1 1 0 1 1-2 0V4H3a1 1 0 0 0-1 1v4.111h20V5a1 1 0 0 0-1-1h-3v2a1 1 0 0 1-2 0V4zm1 0v2-2zm1-2v2h1V2h2a3 3 0 0 1 3 3v18a3 3 0 0 1-3 3H3a3 3 0 0 1-3-3V5a3 3 0 0 1 3-3h2v2h1V1a1 1 0 1 1 2 0v3h1V2h6v2h1V1a1 1 0 0 1 2 0v1zm-1 0V1v1zM7 4v2-2zm0-2V1v1zm15 9.111H2V23a1 1 0 0 0 1 1h18a1 1 0 0 0 1-1V11.111zM7 17a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm5-4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm5-4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2z"
        id="icon-calendar-a"
      />
    </defs>
    <g
      id="icon/secondary/calendar"
    >
      <g
        id="color/earth-400"
        transform="translate(4 3)"
      >
        <mask
          id="icon-calendar-b"
        >
          <use
            xlinkHref="#icon-calendar-a"
          />
        </mask>
        <use
          id="Mask"
          xlinkHref="#icon-calendar-a"
        />
        <g
          id="color/moon-900"
          mask="url(#icon-calendar-b)"
        >
          <path
            d="M0 0h32v32H0z"
            id="Rectangle"
            transform="translate(-4 -3)"
          />
        </g>
      </g>
    </g>
  </svg>
</button>
`;
